### DISCUSSION TITLE
How to have a function called when the uploader is finished creating a base64 image

### DISCUSSION THREAD

**Discussion by jsd3** [xdan/jodit#927] [Q&A] [answered]
I'd like to be able to register a callback function to be called after an image is converted to base64 encoding when the 
uploader option insertImageAsBase64URI is set to true:

`const editor = Jodit.make('#' + targetTextAreaId , {
                            uploader: {
                                insertImageAsBase64URI: true,
                                imagesExtensions: [
                                    "jpg",
                                    "png",
                        ...`

I also want the original image filename - how can I get this?  Thanks so much for any help!

**Comment by xdan** [comment, marked as answer]
Listen for the `afterInsertImage` event — it fires with the inserted `<img>` element right after insertion, including the base64 case:

```js
editor.e.on('afterInsertImage', (img) => {
  // img is the inserted element
});
```

That's your hook to run something once the image is in. (`change` fires too, but `afterInsertImage` hands you the exact node.)
